chia/rpc/wallet_rpc_api.py

```diff
@@ -1699,16 +1701,41 @@ async def get_offer_summary(self, request: Dict[str, Any]) -> EndpointResult:
         offered, requested, infos = offer.summary()
 
         if request.get("advanced", False):
-            return {
+            response = {
                 "summary": {"offered": offered, "requested": requested, "fees": offer.fees(), "infos": infos},
                 "id": offer.name(),
             }
         else:
-            return {
+            response = {
                 "summary": await self.service.wallet_state_manager.trade_manager.get_offer_summary(offer),
                 "id": offer.name(),
             }
 
+        # This is a bit of a hack in favor of returning some more manageable information about CR-CATs
```

Comment:
I think we should do this in a general way since it is expected there will be more special offers with extra info. What is the disadvantage of using plain text data in the offer file? Is there a privacy concern or file size concern? If we can save raw data in the offer file, it will be easier for everyone to read.

Reply:
The offer file format is pretty specifically a spendbundle and changing it now would be a pretty breaking change.  People can make formats on top of the existing format with more metadata, and I have no problem with that.

The problem here is that this information is in the offer file already and it's just a matter of displaying it to the user of the RPC.  I think it's potentially the front-end's responsibility to interpret the proofs checker but that's a big lift which is why we're doing this as a convenience on the backend. 

If more cases like this come up, it will inform me of the more general solution but like I said in the comment, I'm not sure what it is.

Reply:
Yeah that's basically it, the front end needs a much more sophisticated understanding of CLVM than it currently operates with.  Right now it kind of ignores that altogether.  I'm not saying this is good, I think understanding CLVM would be good, but it's a paradigm shift.